deno/lib/types.ts

```diff
@@ -552,6 +553,9 @@ const ulidRegex = /^[0-9A-HJKMNP-TV-Z]{26}$/;
 //   /^([a-f0-9]{8}-[a-f0-9]{4}-[1-5][a-f0-9]{3}-[a-f0-9]{4}-[a-f0-9]{12}|00000000-0000-0000-0000-000000000000)$/i;
 const uuidRegex =
   /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/i;
+const durationRegex =
+  /^P(?!$)(\d+Y)?(\d+M)?(\d+W)?(\d+D)?(T(?=\d)(\d+H)?(\d+M)?(\d+S)?)?$/;
```

Comment:
This does not handle any fractional values, which is particularly useful to be able to use for shorter intervals, revolving around minutes, seconds and fractional seconds.

According to [wikipedia](https://en.wikipedia.org/wiki/ISO_8601#Durations)
> The smallest value used may also have a decimal fraction,[[39]](https://en.wikipedia.org/wiki/ISO_8601#cite_note-39) as in "P0.5Y" to indicate half a year. This decimal fraction may be specified with either a [comma](https://en.wikipedia.org/wiki/Comma) or a [full stop](https://en.wikipedia.org/wiki/Full_stop), as in "P0,5Y" or "P0.5Y"

But I would be happy with only allowing the seconds part to be fractional, possibly only by with a dot (although it seems like the comma is allowed according to the standard), as this is what I have seen for the most part. This would hopefully not allow anything that is not already allowed by the standard (and of course not pass everything that the standard allows for) - but will cover many useful use-cases without too much code changes